File: src/main/java/ru/ioleksiv/telegram/bot/api/annotations/filter/telegram/VenueFilter.java
package ru.ioleksiv.telegram.bot.api.annotations.filter.telegram;

import ru.ioleksiv.telegram.bot.api.annotations.filter.primitive.NotNullFilter;
import ru.ioleksiv.telegram.bot.api.annotations.filter.primitive.StringFilter;
import ru.ioleksiv.telegram.bot.api.model.annotation.AnnotationState;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
@Documented
public @interface VenueFilter {
    AnnotationState status() default AnnotationState.ON;

    LocationFilter location() default @LocationFilter(status = AnnotationState.OFF);

    StringFilter title() default @StringFilter(status = AnnotationState.OFF);

    StringFilter address() default @StringFilter(status = AnnotationState.OFF);

    NotNullFilter foursquareId() default @NotNullFilter(status = AnnotationState.OFF);

    StringFilter foursquareType() default @StringFilter(status = AnnotationState.OFF);

}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/api/annotations/filter/composite/VenueFilter.kt
package io.github.oleksivio.telegram.bot.api.annotations.filter.composite

import io.github.oleksivio.telegram.bot.api.annotations.filter.primitive.NotNullFilter
import io.github.oleksivio.telegram.bot.api.annotations.filter.primitive.StringFilter
import io.github.oleksivio.telegram.bot.api.model.annotation.AnnotationState

@Retention
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@MustBeDocumented
annotation class VenueFilter(val status: AnnotationState = AnnotationState.ON, val validator: Array<String> = [], val location: LocationFilter = LocationFilter(status = AnnotationState.OFF), val title: StringFilter = StringFilter(status = AnnotationState.OFF), val address: StringFilter = StringFilter(status = AnnotationState.OFF), val foursquareId: NotNullFilter = NotNullFilter(status = AnnotationState.OFF), val foursquareType: StringFilter = StringFilter(status = AnnotationState.OFF))
